sample-tests-and-e2e-tests/src/use-effect/use-effect-component-with-props.test.jsx

```diff
@@ -0,0 +1,155 @@
+
+import React from 'react'
+import { shallow } from 'enzyme';
+import { act } from 'react-dom/test-utils';
+
+import { expensiveOperation1, expensiveOperation2, expensiveOperation3, expensiveOperation4 } from './expensiveOperations'
+
+import ComponentWithProps from './use-effect-component-with-props'
+
+jest.mock('./expensiveOperations', () => ({
+    expensiveOperation1: jest.fn(),
+    expensiveOperation2: jest.fn(),
+    expensiveOperation3: jest.fn(),
+    expensiveOperation4: jest.fn(),
+}))
+
+describe('Component with props', () => {
+    const originalString = "A String"
+    const originalArray = ["An", "Array", "Of", "Strings"]
+
+    let wrapper = shallow(<ComponentWithProps strProp={originalString} arrProp={originalArray}/>);
+
+    beforeEach(() => {
+        expensiveOperation1.mockClear();
+        expensiveOperation2.mockClear();
+        expensiveOperation3.mockClear();
+        expensiveOperation4.mockClear();
+
+        wrapper = shallow(<ComponentWithProps strProp={originalString} arrProp={originalArray}/>);
+    })
+
+    it("Should create and call expensive operation 1", () => {
+        expect(expensiveOperation1).toHaveBeenCalledTimes(1);
+    })
+
+    it("Should create and call expensive operation 2", () => {
+        expect(expensiveOperation2).toHaveBeenCalledTimes(1);
+    })
+
+    it("Should create and call expensive operation 3", () => {
+        expect(expensiveOperation3).toHaveBeenCalledTimes(1);
+    })
+
+    it("Should create and call expensive operation 4", () => {
+        expect(expensiveOperation4).toHaveBeenCalledTimes(1);
+    })
+
+    it("Should render a button", () => {
+        expect(wrapper.find('button').length).toBe(1);
+    })
+
+    it('Should render a button containing "Not Clicked"', () => {
```

Comment:
And the same about the two tests below. `useState()` would work fine without this library.

Reply:
I wrote this as I was writing the tests to assert that the known good behavior was indeed testing as expected, before moving onto the unknown behavior.  I'll clean up and remove these tests.